server/gc/src/test/java/org/apache/accumulo/gc/GarbageCollectionTest.java

```diff
@@ -644,4 +768,47 @@ public void bulkImportReplicationRecordsPreventDeletion() throws Exception {
     assertEquals(1, gce.deletes.size());
     assertEquals("hdfs://foo.com:6000/accumulo/tables/2/t-00002/A000002.rf", gce.deletes.get(0));
   }
+
+  /**
+   * Minimal test to show that dir and prevRow are required for valid scan
+   */
+  @Test
+  public void testDirAndPrevRow() throws Exception {
+
+    GarbageCollectionAlgorithm gca = new GarbageCollectionAlgorithm();
+
+    TestGCE gce = new TestGCE();
+    gce.candidates.add("/1636/default_tablet");
+    gce.addDirReference("1636", null, "/default_tablet");
+    gce.addPrevRowReference("1636", null);
+    gca.collect(gce);
+  }
+
+  /**
+   * Show that IllegalState is thrown when no dir entry present in metadata scan.
+   */
+  @Test
+  public void testNoDir() {
+
+    GarbageCollectionAlgorithm gca = new GarbageCollectionAlgorithm();
+
+    TestGCE gce = new TestGCE();
+    gce.candidates.add("/1636/default_tablet");
+    gce.addPrevRowReference("1636", null);
+    assertThrows(IllegalStateException.class, () -> gca.collect(gce));
+  }
+
+  /**
+   * Show that IllegalState is thrown when no prev row present in metadata scan.
+   */
+  @Test
+  public void testNoPrevRow() {
```

Comment:
Need to test two tablets for this, one that is not the last tablet and one that is the last tablet.  Not sure which case this test is testing.

Reply:
Added in 5a893dfbac